x/ecocredit/simulation/genesis.go

```diff
@@ -291,10 +188,45 @@ func getBatchSequence(ctx context.Context, sStore api.StateStore, projectKey uin
 	return seq.NextSequence, nil
 }
 
-func genGenesisState(ctx context.Context, r *rand.Rand, simState *module.SimulationState, ss api.StateStore, ms marketplaceapi.StateStore) error {
+func genGenesisState(ctx context.Context, simState *module.SimulationState, ss api.StateStore, ms marketplaceapi.StateStore) error {
 	accs := simState.Accounts
+	r := simState.Rand
+
 	metadata := simtypes.RandStringOfLength(r, simtypes.RandIntBetween(r, 5, 100))
 
+	// generate core params
+	if err := ss.AllowListEnabledTable().Save(ctx, &api.AllowListEnabled{
+		Enabled: genAllowListEnabled(r),
+	}); err != nil {
+		return err
+	}
+
+	classCreators := genAllowedClassCreators(r, simState.Accounts)
+	for _, creator := range classCreators {
+		if err := ss.AllowedClassCreatorTable().Save(ctx, &api.AllowedClassCreator{
+			Address: creator,
+		}); err != nil {
+			return err
+		}
+	}
+
+	classFees := genCreditClassFee(r)
+
+	fees := []*basev1beta1.Coin{}
+	for i := 0; i < classFees.Len(); i++ {
+		denom := classFees.GetDenomByIndex(i)
+		amount := classFees.AmountOf(denom)
+		fees = append(fees, &basev1beta1.Coin{
+			Denom:  denom,
+			Amount: amount.String(),
+		})
+	}
+	if err := ss.ClassFeesTable().Save(ctx, &api.ClassFees{
```

Comment:
```suggestion
	if err := ss.ClassFeesTable().Insert(ctx, &api.ClassFees{
```

Reply:
We don't have Insert method for singleton tables.